storage/src/test/java/org/apache/kafka/admin/RemoteTopicCrudTest.java

```diff
@@ -0,0 +1,698 @@
+        }
+    }
+
+    @ClusterTest
+    void testCreateRemoteTopicWithInvalidRetentionSize() {
+        // inherited local retention bytes is 1024
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.RETENTION_BYTES_CONFIG, "512"
+        );
+
+        try (var admin = cluster.admin()) {
+            assertThrowsExceptionIgnoringResult(() -> admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig))).all().get());
+        }
+    }
+   
+    @ClusterTest
+    void testCreateCompactedRemoteStorage() {
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true",
+            TopicConfig.CLEANUP_POLICY_CONFIG, "compact"
+        );
+
+        try (var admin = cluster.admin()) {
+            assertThrowsExceptionIgnoringResult(() -> admin.createTopics(
+                List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig))).all().get());
+        }
+    }
+
+    // `remote.log.delete.on.disable` and `remote.log.copy.disable` only works in KRaft mode.
+    @ClusterTests({
+        @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "remote.log.copy.disable", value = "true"),
+            @ClusterConfigProperty(key = "remote.log.delete.on.disable", value = "true")
+        }),
+        @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "remote.log.copy.disable", value = "true"),
+            @ClusterConfigProperty(key = "remote.log.delete.on.disable", value = "false")
+        }),
+        @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "remote.log.copy.disable", value = "false"),
+            @ClusterConfigProperty(key = "remote.log.delete.on.disable", value = "true")
+        }),
+        @ClusterTest(serverProperties = {
+            @ClusterConfigProperty(key = "remote.log.copy.disable", value = "false"),
+            @ClusterConfigProperty(key = "remote.log.delete.on.disable", value = "false")
+        })
+    })
+    void testCreateRemoteTopicWithCopyDisabledAndDeleteOnDisable() throws Exception {
+        var topicConfig = Map.of(
+            TopicConfig.REMOTE_LOG_COPY_DISABLE_CONFIG, cluster.config().serverProperties().get("remote.log.copy.disable"),
+            TopicConfig.REMOTE_LOG_DELETE_ON_DISABLE_CONFIG, cluster.config().serverProperties().get("remote.log.delete.on.disable")
+        );
+
+        try (var admin = cluster.admin()) {
+            var result = admin.createTopics(List.of(new NewTopic(testTopicName, numPartitions, numReplicationFactor).configs(topicConfig)));
+            assertDoesNotThrow(() -> result.all().get(30, TimeUnit.SECONDS));
+        }
+
+        verifyRemoteLogTopicConfigs(topicConfig);
+    }
+
+    // `remote.log.delete.on.disable` only works in KRaft mode.
```

Comment:
ditto: Kafka supports only KRaft mode starting from 4.0, so this comment can be removed.